src/ember-app.js

```diff
@@ -205,10 +208,14 @@ class EmberApp {
    *                                       only
    */
   async getNewApplicationInstance() {
+    const appInstanceRetrievalStartTime = Date.now();
     const queueObject = this._sandboxApplicationInstanceQueue.dequeue();
     const app = await queueObject.item;
 
-    return this.getAppInstanceInfo(app, queueObject.isItemPreBuilt);
+    return this.getAppInstanceInfo(app, {
+      isAppInstancePreBuilt: queueObject.isItemPreBuilt,
+      appInstanceRetrievalTime: Date.now() - appInstanceRetrievalStartTime,
```

Comment:
Instead of doing this here, shouldn't these metrics come from the `queueObject` like `isItemPreBuilt`? It seems a bit of an odd API where we return whether the item was pre built or built on the fly but not the metrics of the time spent for that task. In other words, we are already getting what task was dequeued but not it's timing info. 

 If it isn't prebuilt, it can capture the time to create the sandbox. This way the generator function takes care of creation of the object, determining the state of reterival and metrics. If the intent is to get the time taken to built the sanbox then it should come from as part of the `dequeue` API IMO.